--- a/cpp/DIS/Marking.h
+++ b/cpp/DIS/Marking.h
@@ -20,7 +20,7 @@ protected:
   unsigned char _characterSet; 
 
   /** The characters */
-  char _characters; 
+  char _characters[12]; 
 
 
  public:
@@ -33,8 +33,10 @@ protected:
     unsigned char getCharacterSet() const; 
     void setCharacterSet(unsigned char pX); 
 
-    char getCharacters() const; 
-    void setCharacters(char pX); 
+    char*  getCharacters(); 
+    const char*  getCharacters() const; 
+    void setCharacters( const char*    pX);
+    void setByStringCharacters(const char* pX);
 
 
 virtual int getMarshalledSize() const;
